core/domain/improvements_domain.py

```diff
@@ -137,7 +125,7 @@ def task_id(self) -> str:
         Returns:
             str. The ID of this task.
         """
-        return improvements_models.TaskEntryModel.generate_task_id(
+        return '%s.%s.%d.%s.%s.%s' % (
```

Comment:
Hmm, we are introducing two ways of doing things. How can we make sure that when we change `generate_task_id` we also change it here?

Reply:
Done. I have added a template in `feconf.py`.